app/scripts/keyring-controller.js

```diff
@@ -0,0 +1,572 @@
+const async = require('async')
+const EventEmitter = require('events').EventEmitter
+const encryptor = require('./lib/encryptor')
+const messageManager = require('./lib/message-manager')
+const ethUtil = require('ethereumjs-util')
+const ethBinToOps = require('eth-bin-to-ops')
+const EthQuery = require('eth-query')
+const BN = ethUtil.BN
+const Transaction = require('ethereumjs-tx')
+const createId = require('web3-provider-engine/util/random-id')
+const autoFaucet = require('./lib/auto-faucet')
+const bip39 = require('bip39')
+
+// TEMPORARY UNTIL FULL DEPRECATION:
+const IdStoreMigrator = require('./lib/idStore-migrator')
+
+// Keyrings:
+const SimpleKeyring = require('./keyrings/simple')
+const HdKeyring = require('./keyrings/hd')
+const keyringTypes = [
+  SimpleKeyring,
+  HdKeyring,
+]
+
+module.exports = class KeyringController extends EventEmitter {
+
+  constructor (opts) {
+    super()
+    this.web3 = opts.web3
+    this.configManager = opts.configManager
+    this.ethStore = opts.ethStore
+    this.encryptor = encryptor
+    this.keyringTypes = keyringTypes
+
+    this.keyrings = []
+    this.identities = {} // Essentially a name hash
+
+    this._unconfTxCbs = {}
+    this._unconfMsgCbs = {}
+
+    this.getNetwork = opts.getNetwork
+
+    // TEMPORARY UNTIL FULL DEPRECATION:
+    this.idStoreMigrator = new IdStoreMigrator({
+      configManager: this.configManager,
+    })
+  }
+
+  getState () {
+    const configManager = this.configManager
+    const address = configManager.getSelectedAccount()
+    const wallet = configManager.getWallet() // old style vault
+    const vault = configManager.getVault() // new style vault
+
+    return {
+      seedWords: this.configManager.getSeedWords(),
+      isInitialized: (!!wallet || !!vault),
+      isUnlocked: !!this.key,
+      isConfirmed: true, // AUDIT this.configManager.getConfirmed(),
```

Comment:
Doesn't seem to be hiccuping right now. Tentatively putting it back in until we determine why we set it to true in the first place.